### PR TITLE
Fix backend format to bmv2

### PR DESCRIPTION
**Pull request by sumit7754**
Fixes #4568 

I have fixed the comment formatting in the backend (BMV2) as per the discussions in issue #4567.

Changes made:
- Used triple slashes (///) for commenting functions and classes.
- Used double slashes (//) for comments within the code.

### CODE REVIEW

**Comment by fruffy** [comment]
Do not create a new PR, instead update your old PR.

**Comment by sumit7754** [comment]
/home/runner/work/p4c/p4c/backends/bmv2/common/controlFlowGraph.h:124:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]   how can I solve this error?

**Comment by sumit7754** [comment]
Locally, it's giving the expected outcome.
Done processing /home/sumit/p4c-1/backends/bmv2/common/controlFlowGraph.h

**Comment by fruffy** [comment]
Your clang-format might be mismatched or not formatting correctly. Try to fix it manually.

File: backends/bmv2/common/controlFlowGraph.h [hunk 1]
```diff
@@ -115,21 +115,15 @@ class CFG final : public IHasDbPrint {
...
     class Edge final {
      protected:
         EdgeType type;
         Edge(Node *node, EdgeType type, cstring label) : type(type), endpoint(node), label(label) {}
 
-     public:
-        /**
-         * The destination node of the edge.  The source node is not known by the edge
-         */
+     public:
```
Comment (fruffy) [inline backends/bmv2/common/controlFlowGraph.h:124]: The whitespace is here.

Comment (fruffy) [inline backends/bmv2/common/controlFlowGraph.h:124]:
```suggestion
     public:
```

**Comment by fruffy** [comment]
I just realized this PR only updates the BMv2 back end whereas the other PR updates everything. In that case it is fine to leave this PR open.

**Comment by sumit7754** [comment]
Now failing for this bazel / build_direct but initially it was running fine

**Comment by fruffy** [comment]
> Now failing for this bazel / build_direct but initially it was running fine

This is a failure unrelated to your changes. I will have to take a look.

**Comment by sumit7754** [comment]
sure

**Comment by fruffy** [comment]
Bazel currently does not work because it pulls in a dependency that has major security problems: https://www.openwall.com/lists/oss-security/2024/03/29/4
A backdoor was installed. Quite interesting.

**Comment by sumit7754** [comment]
Thanks for looking into it, @fruffy. Additionally, could you please review the changes I've made and let me know if there's anything that needs attention or could be improved?

**Comment by sumit7754** [comment]
In the process of making these changes, I've discovered some things to avoid. Would it be beneficial to include these insights in our style documentation for future reference?

**Comment by AdarshRawat1** [comment]
> Would it be beneficial to include these insights in our style documentation for future reference?

Absolutely @sumit7754 !! That would be a great asset.

**Comment by sumit7754** [comment]
Sure I will add that

**Comment by sumit7754** [comment]
@fruffy, Can I use the same formatting in the rest of the code, or is there something specific that needs to be changed? If changes are necessary, please let me know, and I'll make the required adjustments.

**Review by fruffy — APPROVED** [review]

**Comment by fruffy** [comment]
> @fruffy, Can I use the same formatting in the rest of the code, or is there something specific that needs to be changed? If changes are necessary, please let me know, and I'll make the required adjustments.

Yes, the formatting you have looks good to me. Thank you for your effort!

**Comment by sumit7754** [comment]
You're welcome, @fruffy. I'm glad the formatting meets your expectations. Regarding the comment formatting of the DPDK backend, I'm contemplating whether to create a new pull request for this task or to integrate it into the existing one.

**Comment by fruffy** [comment]
> I'm contemplating whether to create a new pull request for this task or to integrate it into the existing one.

Let's make it a separate pull request.

**Comment by sumit7754** [comment]
sure

**Comment by sumit7754** [comment]
why it is failing?

**Comment by fruffy** [comment]
Have you tried to build it locally? It might help you debug the problem. 
```<<<<<<< HEAD```

It looks like you used Github's merge tool but left some of the merge markers in your code.

**Comment by sumit7754** [comment]
Yes I have tried to build it locally and also I will remove merge marker in code.

**Pull request closed without merging**